src/main/java/org/jabref/gui/mergeentries/newmergedialog/fieldsmerger/GroupMerger.java

```diff
@@ -21,7 +21,7 @@ public String merge(String groupsA, String groupsB) {
         } else if (StringUtil.isBlank(groupsB)) {
             return groupsA;
         } else {
-            return Arrays.stream((groupsA + GROUPS_SEPARATOR + groupsB).split(GROUPS_SEPARATOR))
+            return Arrays.stream((groupsA + GROUPS_SEPARATOR + groupsB).split("\s*,\s*"))
```

Comment:
You should extract regexes whenever possible to instances of Pattern, e.g. Pattern.compile

Reply:
Ah I see, so it constructs the state machine only once

Reply:
For performance reasons, it doesn't need to create a new object everytime. 
Just use pattern.split()
 JDK11+ also has some fancy new methods 
 https://www.baeldung.com/java-regex-pre-compile